### PR TITLE
Single binary event publisher

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- Unifies the two versions of Eventing Publisher into only once. The new Publisher works with both eventing backends: NATS and BEB.

**Related issue(s)**

### CODE REVIEW

File: components/event-publisher-proxy/cmd/event-publisher-proxy/main.go [hunk 1]
```diff
@@ -1,42 +1,43 @@
...
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/metrics"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/oauth"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/options"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/receiver"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/sender"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/signals"
-	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/subscribed"
+const (
+	BEBBackend  = "beb"
+	NATSBackend = "nats"
```
Comment (marcobebway) [inline components/event-publisher-proxy/cmd/event-publisher-proxy/main.go:15]: Does it make sense to not export those constants outside of package?

File: components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+// Commander implements the Commander interface.
+type Commander struct {
+	envCfg           *env.BebConfig
+	logger           *logrus.Logger
+	metricsCollector *metrics.Collector
+	opts             *options.Options
+}
+
+// NewCommander creates the Commander for publisher to EB.
```
Comment (marcobebway) [inline components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go:31]: `to EB` -> `to BEB`

File: components/event-publisher-proxy/cmd/event-publisher-proxy/main.go (hunk 1 above)
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/main.go:15]: yes, changed

File: components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go (hunk 2 above)
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/beb/beb.go:31]: done

File: components/event-publisher-proxy/cmd/event-publisher-proxy/main.go [hunk 3]
```diff
@@ -49,40 +50,27 @@ func main() {
...
+	// Instantiate configured commander.
+	var commander Commander
+	switch cfg.Backend {
+	case BEBBackend:
+		commander = beb.NewCommander(opts, metricsCollector, logger)
+	case NATSBackend:
+		commander = nats.NewCommander(opts, metricsCollector, logger)
+	default:
+		logger.Fatalf("Invalid publisher backend: %v", cfg.Backend)
+	}
```
[the remark this answers is not in the document]
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/main.go]: Done

File: components/event-publisher-proxy/cmd/event-publisher-proxy/main.go [hunk 4]
```diff
@@ -49,40 +50,27 @@ func main() {
...
 
-	// setup application lister
-	dynamicClient := dynamic.NewForConfigOrDie(k8sConfig)
-	applicationLister := application.NewLister(ctx, dynamicClient)
+	// Init and start the commander.
+	if err := commander.Init(); err != nil {
+		logger.Fatalf("Initialization failed: %s", err)
+	}
```
[the remark this answers is not in the document]
Comment (radufa) [reply components/event-publisher-proxy/cmd/event-publisher-proxy/main.go]: Done

**Review by marcobebway — APPROVED** [review]

**Review by berkanercan — APPROVED** [review]

**Pull request merged**